### ISSUE TITLE
Serialize to json only specific fields

### ISSUE THREAD

**Issue by zahessi** [insightindustry/sqlathanor#60]
Is there a way to serialize a SQLAlchemy model including only specific fields using SQLAthanor? The documentation doesn't mention it, so the only way that I figured out is to filter the outcome manually.

So, this line with sqlathanor

```
return jsonify([user.to_dict() for user in users for k, v in user.to_dict().items() 
                                if k in ['username', 'name', 'surname', 'email']])
```
is equivalent to this one using Marshmallow

`return jsonify(SchemaUser(only=('username', 'name', 'surname', 'email')).dump(users, many=True))`
Once again, is there a built-in method in SQLAthanor to do this?

**Comment by insightindustry** [comment]
@zahessi Yes, absolutely. That is the exact purpose of SQLAthanor. 

The way you do it is even simpler than the code snippet you provided. The process is to set up a configuration when defining your SQLAlchemy model that indicates which fields to serialize (or deserialize, for that matter). Here is the relevant section of documentation that explains how to configure your model’s serialization / deserialization rules: 
https://sqlathanor.readthedocs.io/en/latest/using.html#configuration

This configuration let’s you indicate which fields you want to serialize, and add any ore-processing logic you might want to apply. Once you have configured serialization for your model, you just serialize instances using:

`json_result = user.to_json()`

or

`dict_result = user.to_dict()`

If you want to _change_ the fields that get serialized, you can just modify the configuration on your model instance before calling `to_<FORMAT>()`.

Does this help explain how to serialize specific fields? If you need more help, just let me know and I can provide a more complete explanation.

**Comment by zahessi** [comment]
yeah, I configure the fields that I might want to serialize _in general_ when I initialize my models. But I am asking about the possibility to get only specified fields without modifying the config, like in the marshmallow example higher.

**Comment by insightindustry** [comment]
Ah, gotcha. Then yes, the only way that you can change the list of serialized fields without changing the instance’s configuration is to manually adjust the results of `to_<FORMAT>()`. Your code snippet is one way to do that, although for JSON and YAML you can also supply a custom `serialize_function` which accepts the `dict`, processes it, and serializes to JSON or YAML as appropriate:

    import simplejson as json

    def my_custom_serializer(value):
        filtered_dict = {}
        filtered_dict['username'] = value['username']
        # repeat pattern for other fields

        return json.dumps(filtered_dict)

    json_result = user.to_json(serialize_function = my_custom_serializer)`

Both approaches are effectively the same, but the `serialize_function` approach gives you more flexibility and (I think) easier to read/maintain code.

**Comment by insightindustry** [comment]
Oh, and one other point: you can generalize the `serialize_function` as well. So if you want to give it a list of fields to include, just include them as a keyword argument in `to_json()`:

    def my_custom_serializer(value, **kwargs):
        filter_fields = kwargs.pop(“filter_fields”, None)
        result = {}
        for field in filter_fields:
            result[field] = value.get(field, None)

        return json.dumps(result)

    result = [x.to_json(serialize_funcion = my_custom_serializer, 
                                     filter_fields = ['username', 'name', 'surname', 'email']) for x in users)

**Comment by zahessi** [comment]
ok, thanx for the full answer! By the way, I adore your library) Keep going and luck
